### ISSUE TITLE
White Palace and PoP implementation

### ISSUE THREAD

**Issue by Ijwu** [Ijwu/Archipelago.HollowKnight#34]
We'll need an understanding of the White Palace logic as well as the logic for Path of Pain.

Furthermore, there should be options to disable White Palace and shuffle the White Fragment into the item pool. (We can junk fill the White Fragment vanilla location.)

More info possibly: https://discordapp.com/channels/731205301247803413/959203520521863168/960269754847879199

**Comment by dewiniaid** [comment]
Related: https://github.com/ArchipelagoMW/Archipelago/pull/438

**Comment by dewiniaid** [comment]
If we wanted to add an outside-White-Palace alternate location for KF, it might be doable by pulling from these snippets:

```cs
// Create a new coordinate location in Abyss near the WP entrance
outside_wp = new CoordinateLocation()
{
	x = 120.0,
	y = 17.4
	elevation = 0.0,
	forceShiny = true,
	sceneName = "Abyss_05",
	name = "King_Fragment_Alt"
};

stuff_to_create_and_place_item_goes_here();

new PDBoolCost("dreamNailUpgraded", "Requires Awoken Dream Nail");
```

**Comment by dewiniaid** [comment]
Implemented in main AP 0.3.3
